Task/update fe docs

paulmskim opened the pull request:
Updates to the HTML page for coding standards. Feedback much appreciated.

lucatume commented on docs/code-standards/html.md:
```diff
@@ -42,12 +53,117 @@ Example:
+
+Classes should follow the [Block Element Modifier (BEM)](http://getbem.com/naming/) format for
+naming consistency and ease of targetting in CSS. There are no hard rules for naming in BEM
+(other than what is provided in the link above). However, some ways are better than others. This
```
Here the "some ways are better than others" is expressed, but not defined.
I would add a quick rule of thumb here about how the class names produced in the "good" examples came to be.
E.g. it's not obvious for those less accustomed to BEM, why it's `product-card__description` in place of `product-card__content__description`.

paulmskim replied:
done

lucatume commented on docs/code-standards/html.md:
```diff
@@ -58,22 +174,366 @@ up over multiple lines. In such a case we use the following convention:
-## Anchor tags
+## Semantics
+
+HTML doesn't just display information on a page, it conveys meaning. Semantics are important when writing HTML
```
Would it be worth, here, pointing out that BEM classes !== semantic structure?

paulmskim replied:
done

lucatume commented on docs/code-standards/html.md:
```diff
@@ -58,22 +174,366 @@ up over multiple lines. In such a case we use the following convention:
+
+### Others
+
+There are a variety of other semantic elements that help convey meaning. These include
```
Some examples here using `section` and `article`? 
I mean more from the point of view of our usual approach, e.g. "in TEC we usually represent events with `article` tags, and, in those, we present the date and time in a `section`". 
My example here might be wrong, but I would like to see some distilled knowledge here to guide the first steps.

paulmskim replied:
done

lucatume commented on docs/code-standards/html.md:
````diff
@@ -58,22 +174,366 @@ up over multiple lines. In such a case we use the following convention:
+<img src="/border-line.jpg" alt="" />
+```
+
+### Forms
````
Fields and fieldsets have raised more than one discussion in the past, touch a bit on those here might help.

paulmskim replied:
done

alliuca commented on docs/code-standards/html.md:
````diff
@@ -25,13 +25,24 @@ First and foremost, we make an attempt to adhere to the [WordPress HTML coding s
 
+**Good**
 ```html
 <div id="component">
````
are we aiming to use (BEM) classes instead of ids for styling everywhere?

paulmskim replied:
Yes, this was mainly to address the "code styles" portion above, compared to the "bad" code. But yes, I did miss the `id` here, will change that to class.

alliuca commented on docs/code-standards/html.md:
```diff
@@ -58,22 +174,366 @@ up over multiple lines. In such a case we use the following convention:
+### Forms
+
+Forms are complex HTML structures that require a bit of attention to make them accessible.
+Fortunately, these are fairly easy to implement.
```
it might also be good to set a standard for the submit button, which could be either a `<button>` (defaults to type "submit") or `<input type="submit">`

paulmskim replied:
I honestly don't have a preference to this one, as both are fine and semantically correct to me. Do you have a strong preference one way or another?

alliuca replied:
I don't either, I figured we might set a standard for everyone (unless it makes sense to leave it to other devs to decide on a case by case basis). I personally go for <button> usually and maybe set it to type="button" if it's not supposed to submit the form

borkweb replied:
Likewise, I don't have a strong preference, though I _personally_ prefer `<button>` because it feels more correct and you can do more interesting things with the text/contents of the button.

If we _had_ to pick a preference, I'd lean toward `<button>`.

paulmskim replied:
Yes, this I agree with. I'll add a note on this.

juanfra replied:
Agree!

paulmskim replied:
done

paulmskim commented:
@alliuca @lucatume @borkweb @juanfra thanks for the review and comments, I've addressed the feedback. Let me know if this is good to go and I'll merge it.
